subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -3997,6 +4000,9 @@ int lwm2m_engine_context_close(struct lwm2m_ctx *client_ctx)
 		}
 	}
 
+	for (i = 0, msg = messages; i < len; i++, msg++)
+		memset(msg, 0, sizeof(struct lwm2m_message));
```

Comment:
nit: Zephyr requires brackets even around single-line blocks. Same in new coap functions.